crates/ruma-common/src/events/room/message.rs

```diff
@@ -218,6 +218,81 @@ impl RoomMessageEventContent {
         self
     }
 
+    /// Turns `self` into a [replacement] (or edit) for the message with the given event ID.
+    ///
+    /// This takes the content and sets it in `m.new_content`, and modifies the `content` to include
+    /// a fallback.
+    ///
+    /// If the message that is replaced is a reply to another message, the latter should also be
+    /// provided to be able to generate a rich reply fallback that takes the `body` /
+    /// `formatted_body` (if any) in `self` for the main text and prepends a quoted version of
+    /// `original_message`.
+    #[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/doc/rich_reply.md"))]
+    ///
+    /// # Panics
+    ///
+    /// Panics if `self` has a `formatted_body` with a format other than HTML.
```

Comment:
Why can't this be an error instead?

Reply:
This behavior was copied from the `make_reply_to` constructor.

That's just not a use case we support as formats other than HTML are outside of the spec, and clients know what formats they support for sending so it would be a developer error to use it with another format.